src/metabase/server/middleware/security.clj

```diff
@@ -124,11 +124,52 @@
           "Content-Security-Policy"
           #(format "%s frame-ancestors %s;" % (if allow-iframes? "*" (or (embedding-app-origin) "'none'")))))
 
+(defn- parse-url [url]
+  (let [pattern #"^(?:(https?)://)?([^:/]+)(?::(\d+|\*))?$"
```

Comment:
Hey, it's kind of hard to read regexes (maybe a me-thing), but could you please write a couple tests to show off what sorts of strings this should parse out?